create-docker-action.py

```diff
@@ -0,0 +1,79 @@
+import os
+import pathlib
+
+import yaml
+
+DESCRIPTION = 'description'
+REQUIRED = 'required'
+
+EVENT = os.environ['EVENT']
+REF = os.environ['REF']
+REPO = os.environ['REPO']
+
+
+def set_image(event: str, ref: str, repo: str) -> str:
+    if event == 'pull_request' and 'gh-action-pypi-publish' in repo:
```

Comment:
If the workflow was triggered by a PR to _this repo_ (pypa/gh-action-pypi-publish), then the action should build the Docker image. Else, the action should be using the GHCR Docker image.

This is important because workflows in other repos might run the marketplace action on PRs. For example, they might want to smoke test package uploads using DevPI like this:

```yml
name: Smoke test workflow in a different repo

on:
  pull_request:

env:
  devpi-password: abcd1234
  devpi-username: root
  devpi-port: 3141

jobs:
  smoke-test:
    runs-on: ubuntu-latest
    services:
      devpi:
        image: muccg/devpi
        env:
          DEVPI_PASSWORD: ${{ env.devpi-password }}
        ports:
          - 3141
    timeout-minutes: 2
    steps:
      - name: Check out the action locally
        uses: actions/checkout@v4
      - name: Install the packaging-related tools
        run: python3 -m pip install build twine
      - name: Create the stub package importable directory
        run: mkdir -pv src/test_package
      - name: Populate the stub package `__init__.py`
        run: echo '__version__ = "0.1"' > src/test_package/__init__.py
      - name: Populate the stub package `README.md`
        run: echo "# Test Package" > README.md
      - name: Populate the stub package `pyproject.toml`
        run: echo "$CONTENTS" > pyproject.toml
        env:
          CONTENTS: |
            [build-system]
            requires = [
              "setuptools == 65.6.3",
            ]
            build-backend = "setuptools.build_meta"

            [project]
            name = "test-package"
            version = "0.1"
            readme = "README.md"
      - name: Build the stub package sdist and wheel distributions
        run: python3 -m build
      - name: Register the stub package in devpi
        run: twine register dist/*.tar.gz
        env:
          TWINE_USERNAME: ${{ env.devpi-username }}
          TWINE_PASSWORD: ${{ env.devpi-password }}
          TWINE_REPOSITORY_URL: >-
            http://localhost:${{
              job.services.devpi.ports[env.devpi-port]
            }}/${{
              env.devpi-username
            }}/public/
      - name: Smoke-test the marketplace action
        uses: pypa/gh-action-pypi-publish@unstable/v1
        with:
          user: ${{ env.devpi-username }}
          password: ${{ env.devpi-password }}
          repository-url: http://devpi:${{ env.devpi-port }}/${{ env.devpi-username }}/public/
```

In that case, `github.event_name` would be `pull_request`, but the repo name would be different.

As far as handling PRs to forks of _this repo_ (pypa/gh-action-pypi-publish), if you have some other way of specifying "this repo" that doesn't include `gh-action-pypi-publish`, I'm happy to consider it.

Reply:
> if you have some other way of specifying "this repo" that doesn't include `gh-action-pypi-publish`, I'm happy to consider it.

@br3ndonland yes, the usual check is for the repo owner, not the repo name. So `github.repository_owner == 'pypa'` would work. A more precise check could be `github.repository_owner_id == 178055147` (the ID is taken from https://api.github.com/repos/pypa/gh-action-pypi-publish).

*(https://docs.github.com/en/actions/learn-github-actions/contexts#github-context)*

Reply:
@webknjaz good suggestion. I've updated the Python script and `action.yml` to check the repo owner ID.

Just a small correction - the ID you quoted (`178055147`) is actually `github.repository_id`. `github.repository_owner_id` for PyPA is `647025`. This can also be seen at https://api.github.com/orgs/pypa.

Reply:
@br3ndonland oh, right. I think I meant to suggest that checking the `github.repository_id` would be even more precise since it'd remain immutable in the event of the repository transfer even, but lost track of it. Could you make use of it instead?

Reply:
@webknjaz right, I was thinking the same thing. Updated to use `github.repository_id`.